File: admin-web/src/api/parking-zone.ts
import request from '@/utils/request'
import type { PageResult } from '@/types'

/** 区域视图 */
export interface ParkingZoneVO {
  id: number
  tenantId: number
  lotId: number
  name: string
  tag: string
  level: number
  feeRuleId: number
  totalSpaces: number
  fixedSpaces: number
  tempSpaces: number
  status: number
  managerId: number
  remark: string
  version: number
  createdAt: string
  updatedAt: string
}

/** 区域等级选项 */
export const ZONE_LEVEL_OPTIONS = [
  { label: '普通', value: 1 },
  { label: 'VIP', value: 2 },
  { label: '员工', value: 3 },
]

/** 区域标签选项 */
export const ZONE_TAG_OPTIONS = [
  { label: '普通', value: 'NORMAL' },
  { label: 'VIP', value: 'VIP' },
  { label: '员工', value: 'EMPLOYEE' },
  { label: '装卸', value: 'LOADING' },
  { label: '充电', value: 'CHARGE' },
]

/** 区域状态选项 */
export const ZONE_STATUS_OPTIONS = [
  { label: '启用', value: 1 },
  { label: '禁用', value: 2 },
]

/** 分页查询区域列表 */
export function getParkingZones(params: {
  current?: number
  size?: number
  lotId?: number
  status?: number
}) {
  return request.get<PageResult<ParkingZoneVO>>('/api/v1/parking-zones', params)
}

/** 按车场查询区域列表 */
export function getParkingZonesByLotId(lotId: number) {
  return request.get<ParkingZoneVO[]>(`/api/v1/parking-zones/by-lot/${lotId}`)
}

/** 查询区域详情 */
export function getParkingZone(id: number) {
  return request.get<ParkingZoneVO>(`/api/v1/parking-zones/${id}`)
}

/** 创建区域 */
export function createParkingZone(data: {
  lotId: number
  name: string
  tag?: string
  level?: number
  feeRuleId?: number
  totalSpaces?: number
  fixedSpaces?: number
  status?: number
  managerId?: number
  remark?: string
}) {
  return request.post<ParkingZoneVO>('/api/v1/parking-zones', data)
}

/** 更新区域 */
export function updateParkingZone(id: number, data: Record<string, any>) {
  return request.put<ParkingZoneVO>(`/api/v1/parking-zones/${id}`, data)
}

/** 删除区域（软删除） */
export function deleteParkingZone(id: number) {
  return request.delete<void>(`/api/v1/parking-zones/${id}`)
}

/** 更新区域状态 */
export function updateParkingZoneStatus(id: number, status: number) {
  return request.post<void>(`/api/v1/parking-zones/${id}/status`, undefined, { params: { status } })
}

File: parking-system/src/main/java/com/jushan/platform/modules/parking/entity/ParkingZone.java
package com.jushan.platform.modules.parking.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.jushan.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 区域管理实体。
 * <p>
 * 一个停车场至少包含 1 个区域。
 * tempSpaces = totalSpaces - fixedSpaces，由业务层维护。
 *
 * @author Jushan Platform
 * @since 1.0.0
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("parking_zone")
public class ParkingZone extends BaseEntity {

    private static final long serialVersionUID = 1L;

    /** 所属车场 ID（逻辑外键：parking_lot.id） */
    private Long lotId;

    /** 区域名称 */
    private String name;

    /** 区域标签：NORMAL普通 VIP员工 LOADING装卸 CHARGE充电 支持自定义 */
    private String tag;

    /** 区域等级：1普通 2VIP 3员工 */
    private Integer level;

    /** 收费标准 ID（逻辑外键：fee_rule.id）【预留】 */
    private Long feeRuleId;

    /** 车位总数 */
    private Integer totalSpaces;

    /** 固定车位数 */
    private Integer fixedSpaces;

    /** 临停车位数（= totalSpaces - fixedSpaces） */
    private Integer tempSpaces;

    /** 状态：1启用 2禁用 */
    private Integer status;

    /** 区域负责人 ID（逻辑外键：sys_admin_account.id） */
    private Long managerId;

    /** 备注 */
    private String remark;

    /** 乐观锁版本号 */
    private Integer version;
}
